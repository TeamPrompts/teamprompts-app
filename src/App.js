import React from 'react';
import Button from './components/Button';

function App() {
  return (
    <div className="flex flex-col w-3/4 mx-auto my-12 items-center">
      <h1>Super cool page</h1>
      <Button onClick={() => console.log('I was clicked')}>
        I am a button
      </Button>
    </div>
  );
}

export default App;
