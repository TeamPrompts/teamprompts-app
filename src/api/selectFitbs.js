import { ACTIVE, DESC, FITBS, ID, STATUS } from './constants';
import fitbs from '../mocks/fitbs';
import makeFitb from './makeFitb';
import select from './select';

const { REACT_APP_AIRPLAIN_MODE } = process.env;

function selectFitbs(callback) {
  if (REACT_APP_AIRPLAIN_MODE === 'on') {
    callback(null, fitbs);
  } else {
    select({
      callback,
      map: makeFitb,
      options: {
        filterByFormula: `${STATUS}="${ACTIVE}"`,
        sort: [{ field: ID, direction: DESC }]
      },
      tableName: FITBS
    });
  }
}

export default selectFitbs;
