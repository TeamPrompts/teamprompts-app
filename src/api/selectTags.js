import { ASC, FITBS, NAME, TAGS } from './constants';
import select from './select';

function makeTag(record) {
  return {
    fitbs: record.get(FITBS),
    id: record.id,
    name: record.get(NAME)
  };
}

function selectTags(callback) {
  select({
    callback,
    map: makeTag,
    options: {
      sort: [{ field: NAME, direction: ASC }]
    },
    tableName: TAGS
  });
}

export default selectTags;
