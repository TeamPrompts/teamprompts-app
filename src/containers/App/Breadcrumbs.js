import React from 'react';
import { Link } from 'react-router-dom';
import { BLANK_XS, pageTypes } from '../../constants';

function Breadcrumbs({ pageType, tag }) {
  switch (pageType) {
    case pageTypes.CategoryPage:
      return (
        <div className="mt-2">
          <Link className="text-indigo-700 underline" to="/">
            Home
          </Link>
          {' > '}
          <span className="capitalize">{tag.name}</span>
        </div>
      );
    case pageTypes.HomePage:
    default:
      return <div className="mt-2">&nbsp;</div>;
    case pageTypes.PromptsPage:
      return (
        <div className="mt-2">
          <Link className="text-indigo-700 underline" to="/">
            Home
          </Link>
          {' > '}
          <Link
            className="capitalize text-indigo-700 underline"
            to={`/${tag.slug}`}
          >
            {tag.name}
          </Link>
          {' > '}
          {BLANK_XS}
        </div>
      );
  }
}

export default Breadcrumbs;
