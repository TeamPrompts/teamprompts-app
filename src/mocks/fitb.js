import { SEPARATOR } from '../constants';

export const SOURCE = `Do you know the way to ${SEPARATOR}?`;

const fitb = {
  examples: ['San Jose'],
  id: 'receM4Gus7HH43oLM',
  prompts: ['[place]'],
  source: SOURCE,
  tags: ['recTqD26ElVvXKVad']
};

export default fitb;
