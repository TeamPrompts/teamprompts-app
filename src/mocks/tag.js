import tags from './tags';

const tag = tags.find((tag) => tag.id === 'recTqD26ElVvXKVad');

export default tag;
