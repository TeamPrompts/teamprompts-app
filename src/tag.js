const tag = {
  fitbs: [
    'receM4Gus7HH43oLM',
    'rec99C9lxMl6aSedq',
    'rec590aP1VRIE8f23',
    'recNsahDzKCx0IQPk'
  ],
  id: 'recTqD26ElVvXKVad',
  name: 'retrospectives'
};

export default tag; // INFO: this is a mock
