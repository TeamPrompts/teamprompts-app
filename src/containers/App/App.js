import React, { useEffect, useState } from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import { version } from '../../../package.json';
import selectFitbs from '../../api/selectFitbs';
import selectTags from '../../api/selectTags';
import Footer from '../../components/Footer/Footer';
import Heading from './Heading';
import CategoryPage from './CategoryPage';
import HomePage from './HomePage';
import PromptsPage from './PromptsPage';

// TODO: change to _Provider
function Wrapper({ callback, children, loading }) {
  const [error, setError] = useState();
  const [fitbs, setFitbs] = useState([]);
  const [tags, setTags] = useState([]);
  const [waiting, setWaiting] = useState(true);

  useEffect(() => {
    selectFitbs((error, fitbs) => {
      if (error) {
        setError(error);
      } else {
        setFitbs(fitbs);
      }
    });
  }, []);

  useEffect(() => {
    selectTags((error, tags) => {
      if (error) {
        setError(error);
      } else {
        setTags(tags);
      }
    });
  }, []);

  useEffect(() => {
    if (error) {
      setWaiting(false);
    }
    if (fitbs.length > 0 && tags.length > 0) {
      setWaiting(false);
    }
  }, [error, fitbs.length, tags.length]);

  useEffect(() => {
    if (!waiting) {
      callback(error, { fitbs, tags });
    }
  }, [callback, error, fitbs, tags, waiting]);

  return waiting ? loading : children;
}

function App() {
  const [error, setError] = useState();
  const [fitbs, setFitbs] = useState([]);
  const [tags, setTags] = useState([]);

  return (
    <div className="flex flex-col font-serif items-center max-w-4xl mx-16 sm:mx-32 md:mx-32 lg:mx-32 xl:mx-auto">
      <Wrapper
        callback={(error, { fitbs, tags }) => {
          if (error) {
            setError(error);
          } else {
            setFitbs(fitbs);
            setTags(tags);
          }
        }}
        loading={
          <>
            <div className="mt-2">&nbsp;</div>
            <Heading title="Loading ..." />
          </>
        }
      >
        <Router>
          <Switch>
            <Route
              children={<PromptsPage fitbs={fitbs} tags={tags} />}
              exact={true}
              path="/:slug/:id"
            />
            <Route
              children={<CategoryPage fitbs={fitbs} tags={tags} />}
              exact={true}
              path="/:slug"
            />
            <Route
              children={<HomePage fitbs={fitbs} tags={tags} />}
              default={true}
              path="/"
            />
          </Switch>
        </Router>
      </Wrapper>
      <Footer version={version} />
    </div>
  );
}

export default App;
