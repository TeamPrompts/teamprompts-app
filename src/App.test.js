import { render } from '@testing-library/react';
import React from 'react';
import { create } from 'react-test-renderer';
import App from './App';
import selectFitbs from './api/selectFitbs';
import selectTags from './api/selectTags';
import fitb from './mocks/fitb';
import tag from './mocks/tag';

jest.mock('./api/selectFitbs');
jest.mock('./api/selectTags');

describe('App', () => {
  beforeEach(() => {
    selectFitbs.mockImplementation(callback => callback(null, [fitb]));
    selectTags.mockImplementation(callback => callback(null, [tag]));
  });

  it('loading', () => {
    const tree = create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('w/ fitbs', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('w/ error', () => {
    selectFitbs.mockImplementation(callback => callback(new Error('Oh Noes!')));
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
