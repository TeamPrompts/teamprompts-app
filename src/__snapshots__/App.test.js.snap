// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App click retrospectives 1`] = `
<button
  class="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-900"
  data-testid="retrospectives"
  type="button"
>
  retrospectives (1)
</button>
`;

exports[`App loading 1`] = `
<div
  className="flex flex-col font-serif items-center max-w-4xl mx-16 sm:mx-32 md:mx-32 lg:mx-32 xl:mx-auto"
>
  <h1
    className="font-bold my-8 text-4xl text-gray-900"
  >
    TeamPrompts
  </h1>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-circle-notch fa-w-16 fa-spin text-4xl"
    data-icon="circle-notch"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <small
    className="font-normal my-8 text-gray-700 text-sm"
  >
    v
    0.14.4
  </small>
</div>
`;

exports[`App w/ fitbs & tags 1`] = `
<div>
  <div
    class="flex flex-col font-serif items-center max-w-4xl mx-16 sm:mx-32 md:mx-32 lg:mx-32 xl:mx-auto"
  >
    <h1
      class="font-bold my-8 text-4xl text-gray-900"
    >
      TeamPrompts
    </h1>
    <ul
      class="flex flex-wrap mb-4"
    >
      <li
        class="mx-1 my-1"
      >
        <a
          href="#/all"
        >
          <button
            class="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-900"
            data-testid="all"
            type="button"
          >
            all (1)
          </button>
        </a>
      </li>
      <li
        class="mx-1 my-1"
      >
        <a
          href="#/retrospectives"
        >
          <button
            class="focus:outline-none hover:bg-blue-700 px-4 rounded-full text-white bg-blue-500"
            data-testid="retrospectives"
            type="button"
          >
            retrospectives (1)
          </button>
        </a>
      </li>
    </ul>
    <ul>
      <li>
        <div
          class="border-b border-gray-500 leading-loose py-8 text-lg"
        >
          Do you know the way to 
          <span
            class="inline-flex flex-col leading-none"
          >
            <div
              class="self-center"
              style="display: inline-block;"
            >
              <input
                class=""
                data-testid="input"
                placeholder="________________"
                style="box-sizing: content-box; width: 2px;"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              >
                ________________
              </div>
            </div>
            <label
              class="self-center text-sm"
            >
              [place]
            </label>
          </span>
          ?
          <span
            class="inline-block ml-3 relative"
          >
            <input
              class="hidden"
              data-testid="receM4Gus7HH43oLM"
              id="receM4Gus7HH43oLM"
              type="checkbox"
            />
            <label
              class="cursor-pointer"
              for="receM4Gus7HH43oLM"
            >
              <div
                class="w-6 h-3 bg-gray-400 rounded-full shadow-inner"
              />
              <div
                class="toggle__dot absolute w-4 h-4 bg-gray-300 rounded-full shadow inset-y-0 left-0"
              />
            </label>
          </span>
          <button
            class="bg-blue-500 focus:outline-none font-bold font-mono hover:bg-blue-700 ml-3 px-4 rounded-full text-sm text-white"
            data-testid="copy-button"
            type="button"
          >
            Ctrl+C
          </button>
        </div>
      </li>
    </ul>
    <small
      class="font-normal my-8 text-gray-700 text-sm"
    >
      v
      0.14.4
    </small>
  </div>
</div>
`;

exports[`App w/ fitbs error 1`] = `
<div>
  <div
    class="flex flex-col font-serif items-center max-w-4xl mx-16 sm:mx-32 md:mx-32 lg:mx-32 xl:mx-auto"
  >
    <h1
      class="font-bold my-8 text-4xl text-gray-900"
    >
      TeamPrompts
    </h1>
    <pre>
      {}
    </pre>
    <small
      class="font-normal my-8 text-gray-700 text-sm"
    >
      v
      0.14.4
    </small>
  </div>
</div>
`;

exports[`App w/ tags error 1`] = `
<div>
  <div
    class="flex flex-col font-serif items-center max-w-4xl mx-16 sm:mx-32 md:mx-32 lg:mx-32 xl:mx-auto"
  >
    <h1
      class="font-bold my-8 text-4xl text-gray-900"
    >
      TeamPrompts
    </h1>
    <pre>
      {}
    </pre>
    <small
      class="font-normal my-8 text-gray-700 text-sm"
    >
      v
      0.14.4
    </small>
  </div>
</div>
`;
